File: horner.js
function horner(a, x) {
	result = a[0];
	for (let i = 1; i < a.length; i++) {
		result = result * x + a[i];
	}
	return result;
}

console.log(horner([-0.5, 0.7, -1.1, 0, -0.5, 1], 2));

File: horner.m
% y = -0.5x^5 + 0.7x^4 - 1.1x^3 - 0.5x + 1 x = 2.0

function[w] = horner(a, x)
    w = a(1);
    
    for i = 2:(length(a))
        w = w * x + a(i);
    end
end